tests/test_stl_bind_map.cpp

```diff
@@ -0,0 +1,17 @@
+#include <nanobind/nanobind.h>
+#include <nanobind/stl/bind_map.h>
+#include <nanobind/stl/string.h>
+#include <nanobind/stl/map.h>
```

Comment:
Can you remove `stl/map.h` and `stl/unordered_map.h` from this file? This is likely the source of your error message regarding `NB_OPAQUE` (you imported the default STL map type caster *and* tried to declare bindings for a specific STL map).

Reply:
No, removing them makes the compiler complain (`no template 'map' in namespace 'std'`).

Reply:
You can include `map` from the STL:
```
#include <map>
#include <unordered_map>
```
The nanobind header pulls in type casters that are unwanted in this context.

Reply:
Thanks, that works. Should I do the same with `<string>` instead of `<nanobind/stl/string.h>`?

Reply:
right. You only need `nanobind/stl/..` to have automated conversion of the associated type between Python and C++ (and those automated converters work by always performing a copy).

This means that you would only need `nanobind/stl/string.h` if there is a testcase which returns a string to Python (or takes one as an input).

Reply:
Included `<string>` now instead of `nanobind/stl/string.h`, but the tests are still failing (the maps with non-copyable values).  I don't know what is going wrong there, the branch seems correct - I will look at it later with a fresh mind.